lib/src/radio-group/radio-group.component.ts

```diff
@@ -0,0 +1,139 @@
+import { Component, Input, forwardRef, OnChanges, SimpleChanges, Provider } from "@angular/core";
+import { NG_VALUE_ACCESSOR, ControlValueAccessor } from "@angular/forms";
+
+export interface RadioGroupItem {
+    /** The label or text to be displayed in the list */
+    label: string;
+    /** optional description to be displayed next to the label */
+    description?: string;
+    /** any value which should be tied to the item */
+    value: any;
+    /** optional disabled flag. Will show the item grayed out and disabled */
+    disabled?: boolean;
+    /** The id or the unique key of the item */
+    key: string;
+}
+
+interface UniqueItem {
+    id: string;
+    optionItem: RadioGroupItem;
+    selected: boolean;
+}
+
+interface DisplayItem extends UniqueItem {
+    className: string;
+}
+
+const CUSTOM_RADIOGROUP_CONTROL_VALUE_ACCESSOR: Provider = {
+    provide: NG_VALUE_ACCESSOR,
+    useExisting: forwardRef(() => RadioGroupComponent),
+    multi: true,
+};
+
+@Component({
+    selector: "sebng-radio-group",
+    templateUrl: "./radio-group.component.html",
+    styleUrls: ["./radio-group.component.scss"],
+    providers: [CUSTOM_RADIOGROUP_CONTROL_VALUE_ACCESSOR],
+})
+export class RadioGroupComponent implements ControlValueAccessor, OnChanges {
+    @Input() list: Array<RadioGroupItem>; // TODO: Add support for custom html as well as string labels?
+    @Input() id?: string;
+    @Input() label?: string;
+    @Input() className?: string;
+    @Input() disabled?: boolean = false;
+    @Input() inline?: boolean = false;
+
+    // Placeholders for the callbacks which are later provided
+    // by the Control Value Accessor
+    private onTouchedCallback: () => void;
+    private onChangeCallback: (_: any) => void;
+
+    private _selectedValue: RadioGroupItem = null;
+    public allSelected = null;
+
+    set selectedValue(state: RadioGroupItem) {
+        if (state !== this._selectedValue) {
+            this._selectedValue = state;
+            this.onChangeCallback && this.onChangeCallback(state);
+            this.onTouchedCallback && this.onTouchedCallback();
+
+            this._generateHelperArrays();
+        }
+    }
+
+    get selectedValue() {
+        return this._selectedValue;
+    }
+
+    // HELPER ARRAYS
+    /** array of dropdown item elements with a unique id, the original optionItem and calculated selected property */
+    public uniqueList: Array<UniqueItem> = [];
+    /** Array of dropdown item elements which should be displayed in the current render cycle */
+    public displayList: Array<DisplayItem> = [];
+    /** Array of all dropdown item which are currently selected */
+    public selectedList: Array<RadioGroupItem> = [];
+
+    ngOnChanges(changes: SimpleChanges) {
```

Comment:
ngOnChanges here is angular generic and will executed regardless of which property changed, why don't you use getter and setter functions for list instead ? I suggest moving this to setter function of list.

Reply:
I think he needs it to verify if there are changes, hence this condition
```typescript
if (changes.list && changes.list.previousValue !== changes.list.currentValue) {
```

Reply:
Yousif has a point however I think it's better to move it to a getter, you're right Nuru.